src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('button');
+const messageStart = document.querySelector('.message-start');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const scoreBoard = document.querySelector('.game-score');
+const tableCells = document.querySelectorAll('td');
+const size = 4;
+let score = 0;
+let scoreNumber = [];
+
+const keyCode = {
+  left: 37,
+  up: 38,
+  right: 39,
+  down: 40,
+};
+
+let gameTable = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function isGameTableFull() {
+  for (let i = 0; i < size; i++) {
```

Comment:
Consider using the `some` array method instead of a traditional loop to check if the game table has a zero.